crates/rpc/rpc/src/eth/api/call.rs

```diff
@@ -269,11 +279,77 @@ where
                 }
             }
             // new midpoint
-            mid_gas_limit = (highest_gas_limit + lowest_gas_limit) / 2;
+            mid_gas_limit = ((highest_gas_limit as u128 + lowest_gas_limit as u128) / 2) as u64;
         }
 
         Ok(U256::from(highest_gas_limit))
     }
+
+    pub(crate) async fn create_access_list_at(
+        &self,
+        request: CallRequest,
+        at: Option<BlockId>,
+    ) -> EthResult<AccessList> {
+        let block_id = at.unwrap_or(BlockId::Number(BlockNumberOrTag::Latest));
+        let (mut cfg, block, at) = self.evm_env_at(block_id).await?;
+        let state = self.state_at_block_id(at)?.ok_or_else(|| EthApiError::UnknownBlockNumber)?;
+
+        // we want to disable this in eth_call, since this is common practice used by other node
+        // impls and providers <https://github.com/foundry-rs/foundry/issues/4388>
+        cfg.disable_block_gas_limit = true;
+
+        let mut env = build_call_evm_env(cfg, block, request.clone())?;
+        let mut db = SubState::new(State::new(state));
+
+        let from = request.from.unwrap_or_default();
+        let to = if let Some(to) = request.to {
+            to
+        } else {
+            let nonce = db.basic(from)?.unwrap_or_default().nonce;
+            get_contract_address(from, nonce).into()
+        };
+
+        let initial = request.access_list.clone().unwrap_or_default();
+
+        let precompiles = get_precompiles(&env.cfg.spec_id);
+        let mut inspector = AccessListInspector::new(initial, from, to, precompiles);
+        let (result, _env) = inspect(&mut db, env, &mut inspector)?;
+
+        match result.result {
+            ExecutionResult::Halt { reason, .. } => Err(match reason {
+                Halt::NonceOverflow => InvalidTransactionError::NonceMaxValue,
+                halt => InvalidTransactionError::EvmHalt(halt),
+            }),
+            ExecutionResult::Revert { output, .. } => {
+                Err(InvalidTransactionError::Revert(RevertError::new(output)))
+            }
+            ExecutionResult::Success { .. } => Ok(()),
+        }?;
+        Ok(inspector.into_access_list())
+    }
+}
+
+/// Returns the addresses of the precompiles corresponding to the SpecId.
+fn get_precompiles(spec_id: &SpecId) -> Vec<reth_primitives::H160> {
```

Comment:
this is fine for now, but we need this as part of the EvmProvider trait because precompiles are chain specific.